lib/utils/common.py

```diff
@@ -1740,70 +1780,87 @@ def _format_ns_stop_writes_metrics(
                 continue
 
-            metric = "device_avail_pct"
-            config = "min-avail-pct"
-            usage: str | None = stats.get(metric, None)
-            threshold: str | None = stats.get(config, None)
-
-            if usage is None:
-                metric = "pmem_avail_pct"
-                usage = stats.get(metric, None)
+            metric, usage = _get_first_value_from_dict_with_key(
+                stats,
+                ("data_avail_pct", "device_available_pct", "pmem_available_pct"),
+                default_value=None,
+                return_type=int,
+            )
+            config, threshold = _get_first_value_from_dict_with_key(
+                stats,
+                (
+                    "storage-engine.stop-writes-avail-pct",
+                    "storage-engine.min-avail-pct",
+                ),
+                default_value=None,
+                return_type=int,
+            )
 
-            if usage is not None and threshold is not None:
-                use = int(usage)
-                thresh = int(threshold)
-                sw = _is_stop_writes_cause(use, thresh, stop_writes)
+            if metric and usage is not None and threshold is not None:
+                sw = _is_stop_writes_cause(usage, threshold, stop_writes, invert=True)
                 _create_stop_writes_entry(
                     stop_writes_metrics[node],
                     metric,
-                    use,
+                    usage,
                     sw,
-                    thresh,
+                    threshold,
                     config=config,
                     namespace=ns,
                 )
 
-            metric = "device_used_bytes"
-            config = "max-used-pct"
-            usage: str | None = stats.get(metric, None)
-            bytes_total: str | None = stats.get("device_total_bytes", None)
-            threshold: str | None = stats.get(config, None)
-
-            if usage is None:
-                metric = "pmem_used_bytes"
-                usage = stats.get(metric, None)
-                bytes_total = stats.get("pmem_total_bytes", None)
+            metric, usage = _get_first_value_from_dict_with_key(
+                stats,
+                ("data_used_bytes", "device_used_bytes", "pmem_used_bytes"),
+                default_value=None,
+                return_type=int,
+            )
+            config, threshold = _get_first_value_from_dict_with_key(
+                stats,
+                ("storage-engine.stop-writes-used-pct", "storage-engine.max-used-pct"),
+                default_value=None,
+                return_type=int,
+            )
+            bytes_total: int | float | None = util.get_value_from_dict(
+                stats,
+                ("data_total_bytes", "device_total_bytes", "pmem_total_bytes"),
+                None,
+                return_type=int,
+            )
 
             if usage is not None and threshold is not None and bytes_total is not None:
```

Comment:
Does this also need a check for `metric`? like on line 1799?

Reply:
Should not matter. The other check for metric is also a bit redundant

Reply:
Ill remove the check from the above comment